onnx/shape_inference/implementation.cc

```diff
@@ -179,9 +179,13 @@ static void InferShapesImpl(
       valueTypesByName[vi.name()] = vi.mutable_type();
   }
   for (auto& vi : *g->mutable_output()) {
+    
+    // Save names of output with undefined types
     if (vi.mutable_type() != NULL && !vi.mutable_type()->mutable_tensor_type()->has_elem_type()) {
```

Comment:
I think protobuf implementation will create a type field if it does not exist when we call "mutable_type". Please see: https://developers.google.com/protocol-buffers/docs/reference/cpp-generated#embeddedmessage . I think it is safer to first check "vi.has_type()" before calling mutable_type.